tests/test_analysis.py

```diff
@@ -0,0 +1,95 @@
+import covasim as cv
+import numpy as np
+
+
+# Suite of tests to test basic functionality of the analysis.py file
+# Runtime: 5.004 seconds
+def test_analysis_snapshot():
+    sim = cv.Sim(analyzers=cv.snapshot('2020-04-04', '2020-04-14'))
+    sim.run()
+    snapshot = sim['analyzers'][0]
+    people1 = snapshot.snapshots[0]            # Option 1
+    people2 = snapshot.snapshots['2020-04-04'] # Option 2
+    people3 = snapshot.get('2020-04-14')       # Option 3
+    people4 = snapshot.get(34)                 # Option 4
+    people5 = snapshot.get()                   # Option 5
+    # people3 = []  # uncomment to verify error
+    peoples = [people1, people2, people3, people4, people5]
+    for i, people in enumerate(peoples):
+        optionNum = i+1
+        assert len(people) > 0, f"Option {optionNum} should have more than 0 members"

```

Comment:
I'd suggest `assert len(people1) == len(people4)` or something instead, don't need to test all pairs

Reply:
I'm pretty sure this just just checking to see that each snapshot is populated.  If we set the pop_size in the simulation parameters, is it a safe assumption that every snapshot should have a number of persons  equal to that value.  

So what would you think of this approach?
```
expected_pop_size = 1000
sim = Sim(params{'pop_size': expected_pop_size)
// do stuff
assert len(snapshot_people) == expected_pop_size
```

If dead people are removed from the population, or new people are born / migrated into the population this no longer becomes a valid assumption.

Reply:
@CWiswell-IDM We could do that, my only concern is that the only way this would change is if it changes intentionally (e.g., if we do remove dead people / introduce immigration). I'm concerned about the implications for technical debt for writing tests too closely to how the code is _currently_ written, as opposed to how it _could_ be written. So here, too, I think we would probably be ok to do without the assert, because I can't really think of a way of writing this that won't be brittle. `assert isinstance(people, cv.People)` perhaps? But again my feeling is that this would be more likely to fail intentionally (e.g. if we decide to export a dict for easier processing) than unintentionally...

Reply:
Normally I'm of the opinion that if the return type from a method changes, you should _want_ the tests calling that to fail.  The docstring indicates that we should expect "a snapshot of the sim.people array" from that point in time, but I guess you're right that it doesn't tell us much about what that sim.people array really _is_ under the hood.

I would strongly advocate for codifying this at some point so that the test does something other than "not fail."

Reply:
I see that perspective and I think it's appropriate in cases where a function or method has a definably correct behavior. But I don't feel the snapshop analyzer is the appropriate place to be putting tests for internal properties of the `People` object -- I think those tests, if or when we need them, should be written specifically w.r.t. to the explicit creation of that object.

I haven't kept exact score, but I'd say ~95% of the bugs that the CI tests in Covasim have caught to date have been simply by raising unexpected exceptions ("unknown unknowns"), not by failed asserts or other anticipated problems ("known unknowns").

Reply:
Okay I will modify it so that the parameters of the People object can change, it'll just check that the multiple snapshot options are returning the same thing. I think that would resolve this, but can add more in next commit if not.